From a5927ed663bf4d858c1fea409617ff2bb238e0e5 Mon Sep 17 00:00:00 2001
From: yaozenhu <yaozenhu@realtek.com>
Date: Mon, 14 Nov 2016 10:20:20 +0800
Subject: [PATCH 3/3] [modify] if format change on ac3parse, don't set new
 GST_EVENT_CAPS to OMX. it will reset OMX flow

Change-Id: Ic3cbe735550e7849b95ada2c2bae6f18ea0eb808
---
 gst/audioparsers/gstac3parse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/audioparsers/gstac3parse.c b/gst/audioparsers/gstac3parse.c
index 19a0376..a91bf5d 100644
--- a/gst/audioparsers/gstac3parse.c
+++ b/gst/audioparsers/gstac3parse.c
@@ -698,7 +698,7 @@ gst_ac3_parse_handle_frame (GstBaseParse * parse,
 
   eac = ac3parse->eac; //FIXME
 
-  if (G_UNLIKELY (ac3parse->sample_rate != rate || ac3parse->channels != chans
+  if (G_UNLIKELY ((ac3parse->sample_rate != rate && ac3parse->sample_rate == -1) || (ac3parse->channels != chans && ac3parse->channels == -1)
           || ac3parse->eac != eac)) {
     GstCaps *caps = gst_caps_new_simple (ac3parse->eac ? "audio/x-eac3" : "audio/x-ac3",
         "framed", G_TYPE_BOOLEAN, TRUE, "rate", G_TYPE_INT, rate,
-- 
2.7.4

